src/sentry/api/endpoints/organization_member_invite/details.py

```diff
@@ -140,7 +145,90 @@ def put(
 
         return Response(serialize(invited_member, request.user), status=200)
 
+    def _remove_invite_from_db(
+        self, request: Request, organization: Organization, invited_member: OrganizationMemberInvite
+    ) -> None:
+        audit_data = invited_member.get_audit_log_data()
+        event_name = "INVITE_REMOVE" if invited_member.invite_approved else "INVITE_REQUEST_REMOVE"
+
+        with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+            # also deletes the invite object via cascades
+            invited_member.organization_member.delete()
+
+            self.create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=invited_member.id,
+                event=audit_log.get_event_id(event_name),
+                data=audit_data,
+            )
+
+    def _handle_deletion_by_member(
+        self,
+        request: Request,
+        organization: Organization,
+        invited_member: OrganizationMemberInvite,
+        acting_member: OrganizationMember,
+    ) -> Response:
+        # Members can only delete invitations that they sent
+        if invited_member.inviter_id != acting_member.user_id:
+            return Response({"detail": ERR_MEMBER_INVITE}, status=400)
+
+        self._remove_invite_from_db(
+            request=request, organization=organization, invited_member=invited_member
+        )
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
     def delete(
         self, request: Request, organization: Organization, invited_member: OrganizationMemberInvite
     ) -> Response:
-        raise NotImplementedError
+        # with superuser read write separation, superuser read cannot hit this endpoint
+        # so we can keep this as is_active_superuser
+        if request.user.is_authenticated and not is_active_superuser(request):
+            try:
+                acting_member = OrganizationMember.objects.get(
+                    organization=organization, user_id=request.user.id
+                )
+            except OrganizationMember.DoesNotExist:
+                return Response({"detail": ERR_INSUFFICIENT_ROLE}, status=400)
+
+            has_member_admin_scope = request.access.has_scope("member:admin")
+            if not has_member_admin_scope:
+                can_invite_members = request.access.has_scope("member:invite")
+                if can_invite_members:
+                    return self._handle_deletion_by_member(
+                        request, organization, invited_member, acting_member
+                    )
+                return Response({"detail": ERR_INSUFFICIENT_SCOPE}, status=400)
+            else:
+                can_manage = roles.can_manage(acting_member.role, invited_member.role)
```

Comment:
I think you can move this further down the method, and do all the low-hanging fruit checks and return early from there. It will make this section a lot less nested.